specifyweb/specify/views.py

```diff
@@ -456,6 +456,16 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
         except (ValueError):
             continue
 
+        # Handle case of updating a large amount of agent ids in the audit logs.
+        # Fix by optimizing the query by consolidating it here
+        if model_name.lower() == 'agent' and table_name.lower() == 'spauditlog':
+            for field_name_id in ['createdbyagent_id', 'modifiedbyagent_id']:
+                query: Q = Q(**{field_name_id: old_model_ids[0]})
```

Comment:
can mypy infer the type automatically?

```suggestion
                query = Q(**{field_name_id: old_model_ids[0]})
```